syntax = "proto3";

package bluetooth.hal.facade;

import "google/protobuf/empty.proto";
import "facade/common.proto";

// The HCI HAL transportation layer certification API surface definition

service HciTransportation {
  rpc SetLoopbackMode(LoopbackModeSettings) returns (google.protobuf.Empty) {}
  rpc SendHciCommand(HciCmdPacket) returns (google.protobuf.Empty) {}
  rpc SendHciAcl(HciAclPacket) returns (google.protobuf.Empty) {}
  rpc SendHciSco(HciScoPacket) returns (google.protobuf.Empty) {}

  rpc FetchHciEvent(bluetooth.facade.EventStreamRequest) returns (stream HciEvtPacket) {}
  rpc FetchHciAcl(bluetooth.facade.EventStreamRequest) returns (stream HciAclPacket) {}
  rpc FetchHciSco(bluetooth.facade.EventStreamRequest) returns (stream HciScoPacket) {}
}

message LoopbackModeSettings {
  bool enable = 1;
}

message HciEvtPacket {
  bytes payload = 1;
}

message HciCmdPacket {
  bytes payload = 1;
}

message HciAclPacket {
  bytes payload = 1;
}

message HciScoPacket {
  bytes payload = 1;
}
